tools/src/bin/collect.rs

```diff
@@ -273,15 +312,46 @@ struct QueryOptions {
     current_batch: bool,
 }
 
+#[derive(Debug, Derivative, PartialEq, Eq, Subcommand)]
+enum Subcommands {
+    /// Create a new collection job and poll it to completion
+    ///
+    /// This is the default action when no subcommand is provided
+    Run,
+    /// Initialize a new collection job
+    ///
+    /// Outputs the newly generated collection job ID to stdout
+    NewJob,
```

Comment:
This required introduction of a new method for the collector client `Collector::start_collection_with_id()`, since the existing `start_collection()` randomly generates the ID for us.

I wonder if for the next round of breaking changes if we should replace `start_collection()` to have the same behavior as `start_collection_with_id()`.

Reply:
There is also an argument to be had about having job deletion support, but I will leave that for a future PR to keep this one small.

Reply:
I like the idea of `Collector`'s interface changing to always take a collection job ID with the next round of breaking changes -- it's easy enough to generate a random ID that I don't think this is much of a burden on consumers of the library.